From 533a78f3de44a39ec91e7eb1891822847cd9582b Mon Sep 17 00:00:00 2001
From: Liron Himi <lironh@marvell.com>
Date: Mon, 10 May 2021 15:46:11 +0300
Subject: [PATCH 1526/1921] net: mvpp2: apply mac dst on MUSDK case

currently without 'if up' before running
MUSDK app, mac destination is not set.
this patch apply the mac destination also
in MUSDK case

Change-Id: Iba26e7d9fa8d042e708c071227b4a89a6ae467ca
Signed-off-by: Liron Himi <lironh@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/51466
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
(cherry picked from commit 14223dece1172f5b95bb8a9a07d065f340f34f34)
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/51533
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 8032464f6efb..1bc813618b6c 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -4895,14 +4895,15 @@ static int mvpp2_open(struct net_device *dev)
 		return err;
 	}
 
-	if (port->flags & MVPP22_F_IF_MUSDK)
-		goto skip_musdk_parser;
-
 	err = mvpp2_prs_mac_da_accept(port, dev->dev_addr, true);
 	if (err) {
 		netdev_err(dev, "mvpp2_prs_mac_da_accept own addr failed\n");
 		return err;
 	}
+
+	if (port->flags & MVPP22_F_IF_MUSDK)
+		goto skip_musdk_parser;
+
 	err = mvpp2_prs_tag_mode_set(port->priv, port->id, MVPP2_TAG_TYPE_MH);
 	if (err) {
 		netdev_err(dev, "mvpp2_prs_tag_mode_set failed\n");
-- 
2.31.1

